cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunBallerinaPreBuildToolsTask.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.cli.task;
+
+import io.ballerina.cli.tool.BuildToolRunner;
+import io.ballerina.cli.utils.FileUtils;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectException;
+import io.ballerina.projects.ToolContext;
+import io.ballerina.toml.api.Toml;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Collection;
+import java.util.List;
+import java.util.ServiceLoader;
+
+import static io.ballerina.cli.launcher.LauncherUtils.createLauncherException;
+import static io.ballerina.projects.PackageManifest.Tool;
+
+/**
+ * Task for running tools integrated with the build.
+ */
+public class RunBallerinaPreBuildToolsTask implements Task {
+
+    private final PrintStream outStream;
+
+    public RunBallerinaPreBuildToolsTask(PrintStream out) {
+        this.outStream = out;
+    }
+
+    @Override
+    public void execute(Project project) {
+        Collection<Diagnostic> toolDiagnostics = project.currentPackage().manifest().diagnostics().diagnostics();
+        boolean hasTomlErrors = project.currentPackage().manifest().diagnostics().hasErrors();
+        if (hasTomlErrors) {
+            toolDiagnostics.forEach(outStream::println);
+            throw createLauncherException("ballerina.toml validation for pre build tool execution contains errors");
```

Comment:
We print a generic error for all failures: `compilation contains errors` Let's stick to that.